OsmAnd/src/net/osmand/plus/routing/RoutingHelper.java

```diff
@@ -260,6 +262,17 @@ public LatLon getFinalLocation() {
 		return finalLocation;
 	}
 
+	public void cacheStartLocation(Location nextStartLocation) {
+		LatLon start = new LatLon(nextStartLocation.getLatitude(), nextStartLocation.getLongitude());
+		if (lastStartLocation == null) {
+			lastStartLocation = new LatLon(nextStartLocation.getLatitude(), nextStartLocation.getLongitude());
+			app.getMapViewTrackingUtilities().detectDrivingRegion(lastStartLocation);
+		} else if (MapUtils.getDistance(start, lastStartLocation) > 100000) {
```

Comment:
Constant ?

Reply:
Could be 1 line "lastStartLocation == null || MapUtils.getDistance(start, lastStartLocation) < CACHE_RADIUS )

Reply:
fixed